packages/atclient/include/atclient/cjson.h

```diff
@@ -1,6 +1,10 @@
 #ifndef ATCLIENT_CJSON_H
 #define ATCLIENT_CJSON_H
 
+#if defined(CONFIG_IDF_TARGET_ESP32)
+#include <cjson.h>
+#else
 #include <cJSON.h>
+#endif
```

Comment:
`atclient/cjson.h` has a preprocessor check if we're ESP32

Then, the `atclient`'s CMakeLists.txt will now do `REQUIRES json` for cJSON stuff.